### ISSUE TITLE
Output format option for lint

### ISSUE THREAD

**Issue by tjkirch** [tamasfe/taplo#328]
It'd be great to have an option to `taplo lint` to allow other output formats, like JSON.

This would allow for easier integration into other tools; for example, I'm trying to get taplo linting working with neovim via [null-ls](https://github.com/jose-elias-alvarez/null-ls.nvim), which only supports taplo formatting today.  The linting output is great for humans, but hard to parse for tools like this.  (null-ls acts as an LSP server, with neovim as the client, so I don't think taplo-lsp fits into the picture.)

I see there's JSON output for `taplo get` so it seems like the pieces we'd need are there.

**Comment by tjkirch** [comment]
I took a look through taplo and it seems like this change would have a fairly small scope.

1. Add an `output_format` argument to `LintCommand` in `taplo-cli/src/args.rs`, just like for `GetCommand`, but referring to a new enum.  (`OutputFormat`, used by `GetCommand`, has a `Value` variant that doesn't particularly make sense for `lint`.)  The new enum could have `Text` and `Json` variants, and perhaps `Toml`, but I think most tool integration would want JSON.
2. Change the `print_*` methods in `taplo-cli/src/printing.rs` to take the output format into account, just like `execute_get` does in `taplo-cli/src/commands/queries.rs` for `get`.  I only care about syntax errors (`print_parse_errors`) which is the simplest method; I'm not as familiar with the use cases for semantic and schema errors.

One thing I haven't figured out yet is how to get a row/col (and, ideally, ending row/col) from the `TextRange` stored in the `parser::Error`s that are available in the print methods, which seem to store only an index into the full text.  Other than that, it seems somewhat straightforward.

I'm curious whether something like this has already been considered and discarded, or if this plan could work.

**Comment by nemethf** [comment]
I'm not familiar with neovim or null-ls, so I wonder what additional features you gain by using taplo via null-ls instead of using taplo directly with neovim?

**Comment by tjkirch** [comment]
Thanks for the tip!  I just found that nvim-lspconfig has a [config](https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/taplo.lua) for taplo built-in which uses taplo's LSP subcommand.  I went down the route of null-ls because null-ls ships with a taplo formatting config built-in, and I thought it could do the linting at the same time.  But nvim-lspconfig is required in any case, so using its config makes sense.  Thanks again.

**Comment by peterhoeg** [comment]
I thought I better revive this rather than open a new issue as I still think the original request makes sense.

If the editor of choice supports LSP, most people will probably just use taplo's lsp mode, but not everybody does that so having a format that could be parsed easily by the editor plugins would be great.

It could be JSON which would of course allow much more data or just a simple `filename:line:column: and some message goes here` format that could be picked up using standard regexp matchers.
